### DISCUSSION TITLE
AdapterService signature does not fit ServiceMethods signature

### DISCUSSION THREAD

**Discussion by mariusheine** [feathersjs/feathers#2340] [Q&A] [answered]
The `AdapterService` of the `adapter-commons` has different service method signatures than expected by implementing `ServiceMethods`.

https://github.com/feathersjs/feathers/blob/d332eaca9945aab0c222413a7e46cec1c0601648/packages/adapter-commons/src/service.ts#L114

This causes the call `app.use(path, adapterServiceInstance)` to result in a typescript error. The main problem seems to be the intersection with `Paginated<T>`.

Why does not the find method of `ServiceMethods` declare the return type as `Promise<T[] | Paginated<T>>`. Then we would have no clash as your docs about [Pagination](https://docs.feathersjs.com/api/databases/common.html#pagination) say that each service should handle it properly. Or did I get something wrong?

Why are the adapter-commons and service interfaces not on the same basis?

I would like to help but I just need somehow to understand the thoughts or history behind the differences,

Thx in advance!

**Comment by daffl** [comment, marked as answer]
The  declarations in the main repository are for the upcoming version 5 and include breaking changes from the current version 4. The adapters (other than `@feathersjs/adapter-memory`) have not been updated accordingly yet.

**Reply by mariusheine** [reply]
Alright. Thx for your fast answer. We are already using the v5 pre version. If I have time and find a good solution I might propose some PRs. 

Go on with the good work.

**Reply by daffl** [reply]
There might be some things that can be done for better backwards compatibility but even something like moving the `Paginate` type out of the main repo to a better spot (since it is adapter specific) is going to break things (even though the services still work exactly the same).
